Module fails to resolve with `--experimental-resolve-tsconfig-paths` after update from 0.14 to 0.15

After updating to `0.15.1` from `0.14` the `--experimental-resolve-tsconfig-paths` option stopped working for me.

Here is the stack trace:

```
Cannot find module '[project]/tsconfig.json/src/db'
Require stack:
- [project]/.config/kysely.config.ts

    at Function._resolveFilename (node:internal/modules/cjs/loader:1365:15)
    at Function.resolve (node:internal/modules/helpers:145:19)
    at jitiResolve (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:190064)
    at jitiRequire (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:192130)
    at import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202951)
    at .config/kysely.config.ts:4:17
    at async Function.import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202945)
    at async resolveConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:376:18)
    at async loadConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:168:23)
    at async getConfig (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.8.3/node_modules/kysely-ctl/dist/bin.js:275:24)
    at async getConfigOrFail (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.8.3/node_modules/kysely-ctl/dist/bin.js:312:18)
    at async usingMigrator (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.8.3/node_modules/kysely-ctl/dist/bin.js:601:18)
    at async Object.run (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.8.3/node_modules/kysely-ctl/dist/bin.js:670:5)
    at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:316:16)
    at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
    at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
    at async runMain (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:445:7)
    at async Object.parse (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.8.3/node_modules/kysely-ctl/dist/bin.js:1287:7)
```

In `0.14` when the module fails to resolve I get the following (when I purposely create a bad path):

```
Cannot find module '[project]/src/db-fail'
...
```

So it looks like it's including `tsconfig.json` in the resolved path.

Reviewing the following it kind of looks it's using a filename instead of a directory:

https://github.com/kysely-org/kysely-ctl/pull/231/files#diff-ec9b5d00f83be5e9bca1bc785864de8b02b1e863903ccd5d1c94434c612091d7R56-R57

If I run the following in the terminal it does include the `tsconfig.json` file in the resolved path:

```
> const { resolve } = require('pathe')
undefined
> resolve('/[project]/tsconfig.json', '.', 'test')
'/[project]/tsconfig.json/test'
```

That said, just my random speculation from looking at the code quickly and I might be missing something obvious.

Related files:

<details>
<summary><code>tsconfig.json</code></summary>

```json
{
  "compilerOptions": {
    "target": "ES2017",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./src/*"],
      "@/tests/*": ["./tests/*"]
    }
  },
  "include": ["next-env.d.ts", "custom.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts", ".config/*.ts"],
  "exclude": ["node_modules"]
}
```
</details>

<details>
<summary><code>.config/kysely.config.ts</code></summary>

```ts
import { PostgresDialect } from 'kysely';
import { defineConfig } from 'kysely-ctl';

import { pool } from '@/db';

export default defineConfig({
  dialect: new PostgresDialect({ pool }),
  migrations: {
    migrationFolder: 'migrations',
  },
  plugins: [],
});
```
</details>

## Work around

Add `"baseUrl": "./"` to `tsconfig.json`.

Here is the full output:

```sh
$ CI=true pnpm kysely migrate latest --experimental-resolve-tsconfig-paths --debug

[debug] cwd /tmp/debug
[debug] { rawArgs: [ '--experimental-resolve-tsconfig-paths', '--debug' ],
  args: { _: [], 'experimental-resolve-tsconfig-paths': true, debug: true },
  data: undefined,
  cmd:
   { meta:
      { name: 'latest', description: 'Update the database schema to the latest version' },
     args:
      { cwd: [Object],
        debug: [Object],
        environment: [Object],
        'experimental-resolve-tsconfig-paths': [Object],
        'no-filesystem-caching': [Object],
        'no-outdated-check': [Object],
        'no-transaction': [Object] },
     run: [AsyncFunction: run] } } []
[debug] /tmp/debug/tsconfig.json { compilerOptions:
   { target: 'ES2017',
     lib: [ 'dom', 'dom.iterable', 'esnext' ],
     allowJs: true,
     skipLibCheck: true,
     strict: true,
     noEmit: true,
     esModuleInterop: true,
     module: 'esnext',
     moduleResolution: 'bundler',
     resolveJsonModule: true,
     isolatedModules: true,
     jsx: 'preserve',
     incremental: true,
     plugins: [ [Object] ],
     paths: { '@/*': [Array], '@/tests/*': [Array] } },
  include:
   [ 'next-env.d.ts',
     'custom.d.ts',
     '**/*.ts',
     '**/*.tsx',
     '.next/types/**/*.ts',
     '.config/*.ts' ],
  exclude: [ 'node_modules' ] }
[debug] jitiOptions { alias:
   { '@': '/tmp/debug/tsconfig.json/src',
     '@/tests': '/tmp/debug/tsconfig.json/tests' },
  debug: true,
  fsCache: false,
  jsx: true,
  tryNative: false }
[jiti] [init] version: 2.5.1 module-cache: true fs-cache: false rebuild-fs-cache: false interop-defaults: true
[jiti] [transpile] [esm] ./.config/kysely.config.ts (17.438ms)
[jiti] [native] [import] ./node_modules/.pnpm/kysely@0.28.5/node_modules/kysely/dist/esm/index.js
[jiti] [native] [import] ./node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/index.js
[error] Cannot find module '/tmp/debug/tsconfig.json/src/db'
Require stack:
- /tmp/debug/.config/kysely.config.ts
  at Function._resolveFilename (node:internal/modules/cjs/loader:1365:15)
  at Function.resolve (node:internal/modules/helpers:145:19)
  at jitiResolve (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:190064)
  at jitiRequire (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:192130)
  at import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202951)
  at .config/kysely.config.ts:4:17
  at async Function.import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202945)
  at async resolveConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:376:18)
  at async loadConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:168:23)
  at async getConfig (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:275:24)
  at async getConfigOrFail (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:312:18)
  at async usingMigrator (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:601:18)
  at async Object.run (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:670:5)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:316:16)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
  at async runMain (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:445:7)
  at async Object.parse (node_modules/.pnpm/kysely-ctl@0.15.1_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:1287:7) 

[error] Cannot find module '/tmp/debug/tsconfig.json/src/db'
Require stack:
- /tmp/debug/.config/kysely.config.ts
```

@igalklebanov So [v0.15.2](https://github.com/kysely-org/kysely-ctl/releases/tag/v0.15.2) now works without a `baseUrl`, but doesn't work if you don't have an `extends` in your `tsconfig.json`.

Looking at the updated `getTSConfigs` it looks like [config](https://github.com/kysely-org/kysely-ctl/blob/b5d2bc7a8e1a9cf40d93073830001589ea408463/src/utils/tsconfig.mts#L33-L37) will be empty if extended is undefined/empty. This will mean `configs.length` will be false and the aliases won't get set because the function [returns early](https://github.com/kysely-org/kysely-ctl/blob/b5d2bc7a8e1a9cf40d93073830001589ea408463/src/utils/jiti.mts#L38-L42).

<details>
<summary>Updated debug</summary>

```
[debug] cwd /tmp/debug
[debug] { rawArgs: [ '--experimental-resolve-tsconfig-paths', '--debug' ],
  args: { _: [], 'experimental-resolve-tsconfig-paths': true, debug: true },
  data: undefined,
  cmd:
   { meta:
      { name: 'latest', description: 'Update the database schema to the latest version' },
     args:
      { cwd: [Object],
        debug: [Object],
        environment: [Object],
        'experimental-resolve-tsconfig-paths': [Object],
        'no-filesystem-caching': [Object],
        'no-outdated-check': [Object],
        'no-transaction': [Object] },
     run: [AsyncFunction: run] } } []
[debug] extended undefined
[debug] tsconfig {
  "compilerOptions": {
    "target": "ES2017",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": [
        "./src/*"
      ],
      "@/tests/*": [
        "./tests/*"
      ]
    }
  },
  "include": [
    "next-env.d.ts",
    "custom.d.ts",
    "**/*.ts",
    "**/*.tsx",
    ".next/types/**/*.ts",
    ".config/*.ts"
  ],
  "exclude": [
    "node_modules"
  ]
}
[debug] jitiOptions { alias: {}, debug: true, fsCache: false, jsx: true, tryNative: false }
[jiti] [init] version: 2.5.1 module-cache: true fs-cache: false rebuild-fs-cache: false interop-defaults: true
[jiti] [transpile] [esm] ./.config/kysely.config.ts (17.332ms)
[jiti] [native] [import] ./node_modules/.pnpm/kysely@0.28.5/node_modules/kysely/dist/esm/index.js
[jiti] [native] [import] ./node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/index.js
[error] Cannot find module '@/db'
Require stack:
- /tmp/debug/.config/kysely.config.ts
  at Function._resolveFilename (node:internal/modules/cjs/loader:1365:15)
  at Function.resolve (node:internal/modules/helpers:145:19)
  at jitiResolve (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:190064)
  at jitiRequire (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:192130)
  at import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202951)
  at .config/kysely.config.ts:4:17
  at async Function.import (node_modules/.pnpm/jiti@2.5.1/node_modules/jiti/dist/jiti.cjs:1:202945)
  at async resolveConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:376:18)
  at async loadConfig (node_modules/.pnpm/c12@3.2.0/node_modules/c12/dist/shared/c12.DDjD4HmS.mjs:168:23)
  at async getConfig (node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:275:24)
  at async getConfigOrFail (node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:312:18)
  at async usingMigrator (node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:601:18)
  at async Object.run (node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:670:5)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:316:16)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
  at async runCommand (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:307:11)
  at async runMain (node_modules/.pnpm/citty@0.1.6/node_modules/citty/dist/index.mjs:445:7)
  at async Object.parse (node_modules/.pnpm/kysely-ctl@0.15.2_kysely@0.28.5_typescript@5.9.2/node_modules/kysely-ctl/dist/bin.js:1287:7) 

[error] Cannot find module '@/db'
Require stack:
- /tmp/debug/.config/kysely.config.ts
```

</details>

--- a/examples/node-esm-tsconfig-paths-no-baseurl/tsconfig.json
+++ b/examples/node-esm-tsconfig-paths-no-baseurl/tsconfig.json
@@ -4,6 +4,5 @@
 		 	"@/*": ["./src/*"]
 		}
 	},
-	"extends": "../../tsconfig.base.json",
 	"include": [".config/**/*", "src", "migrations"]
 }

--- a/src/utils/jiti.mts
+++ b/src/utils/jiti.mts
@@ -50,7 +50,7 @@ async function getJitiAliasFromTSConfig(): Promise<Record<string, string>> {
 			return {}
 		}
 
-		const { baseUrl = '.' } = (merged.compilerOptions || {}) as CompilerOptions
+		const { baseUrl = '.' } = (merged?.compilerOptions || {}) as CompilerOptions
 		// biome-ignore lint/style/noNonNullAssertion: paths != null => filepath != null
 		const dirpath = dirname(filepath!)
 

--- a/src/utils/tsconfig.mts
+++ b/src/utils/tsconfig.mts
@@ -20,11 +20,12 @@ const cache = new TSConfckCache<
 
 export async function getTSConfigs(): Promise<{
 	configs: TSConfigWithPath[]
-	merged: TSConfig
+	merged: TSConfig | undefined | null
 }> {
-	const { extended, tsconfig } = await parse(join(getCWD(), 'index.ts'), {
-		cache,
-	})
+	const { extended, tsconfig, tsconfigFile } = await parse(
+		join(getCWD(), 'index.ts'),
+		{ cache },
+	)
 
 	consola.debug('extended', JSON.stringify(extended, null, 2))
 	consola.debug('tsconfig', JSON.stringify(tsconfig, null, 2))
@@ -34,7 +35,7 @@ export async function getTSConfigs(): Promise<{
 			extended?.map((result) => ({
 				filepath: result.tsconfigFile,
 				tsconfig: result.tsconfig,
-			})) || [],
+			})) || (tsconfig ? [{ filepath: tsconfigFile, tsconfig }] : []),
 		merged: tsconfig,
 	}
 }

